Packs/Coralogix/Integrations/Coralogix.py

```diff
@@ -0,0 +1,447 @@
+    search_api_url = urljoin(demisto.params()['esapi_url'].strip(), '/*/_search')
+
+    tag_application_name = demisto.params()['app_name'].strip()
+    tag_subsystem_name = demisto.params()['subsystem_name'].strip()
+
+    if 'incidents_query' in demisto.params() and demisto.params()['incidents_query'] is not None:
+        incidents_query = demisto.params()['incidents_query'].strip()
+    else:
+        incidents_query = "alert_type:\"53d222e2-e7b2-4fa6-80d4-9935425d47dd\""
+    if 'incidents_application_name' in demisto.params() and demisto.params()['incidents_application_name'] is not None:
+        incidents_application_name = demisto.params()['incidents_application_name'].strip()
+    else:
+        incidents_application_name = ""
+    if 'incidents_subsystem_name' in demisto.params() and demisto.params()['incidents_subsystem_name'] is not None:
+        incidents_subsystem_name = demisto.params()['incidents_subsystem_name'].strip()
+    else:
+        incidents_subsystem_name = ""
+    if 'incidents_severity' in demisto.params() and demisto.params()['incidents_severity'] is not None:
+        incidents_severity = demisto.params()['incidents_severity'].strip()
+    else:
+        incidents_severity = ""
+    if 'incident_description_field' in demisto.params() and demisto.params()['incident_description_field'] is not None:
+        incidents_name_field = demisto.params()['incident_description_field'].strip()
+    else:
+        incidents_name_field = "alert_name"
+
+    LOG(f'Command being called is {demisto.command()}')
+    try:
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(cgx_endpoint_url=search_api_url, cgx_private_key=private_key)
+            demisto.results(result)
+        elif demisto.command() == 'fetch-incidents' or demisto.command() == 'clx_fetch_incidents':
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            if len(incidents_query.strip()) > 0 and len(incidents_name_field.strip()) > 0:
+                incidents = fetch_incidents(
+                    cgx_private_key=private_key,
+                    cgx_endpoint_url=search_api_url,
+                    incidents_base_query=incidents_query,
+                    application_name=incidents_application_name,
+                    subsystem_name=incidents_subsystem_name,
+                    severity=incidents_severity,
+                    incidents_name_field=incidents_name_field
+                )
+                if demisto.command() == 'clx_fetch_incidents':
+                    demisto.results(incidents)
+                else:
+                    demisto.incidents(incidents)
+            else:
+                if demisto.command() == 'clx_fetch_incidents':
+                    if len(incidents_query.strip()) == 0:
```

Comment:
Just another part of my paranoidic programming style - I did that to be extra safe just in case I (or someone else from our company) would choose to re-use this code elsewhere...